inc/saidot1brport.h

```diff
@@ -0,0 +1,145 @@
+/*
+* Copyright (c) 2015 Dell Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may
+* not use this file except in compliance with the License. You may obtain
+* a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*
+* THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR
+* CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+* LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+* FOR A PARTICULAR PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
+*
+* See the Apache Version 2.0 License for specific language governing
+* permissions and limitations under the License.
+*
+* @file saidot1brport.h
+*
+* @brief This module defines SAI API for IEEE 802.1BR Port attributes.
+************************************************************************/
+
+#if !defined (__SAIDOT1BRPORT_H)
+#define __SAIDOT1BRPORT_H
+
+#include "saitypes.h"
+#include "saistatus.h"
+
+/** \defgroup SAIDOT1BRPORT SAI - 802.1BR Port specific public APIs and datastructures.
+ *
+ *  \{
+ */
+
+/**
+ * @brief Attribute data for SAI_DOT1BR_PORT_ATTR_TYPE
+ */
+typedef enum _sai_dot1br_port_type_t
+{
+    SAI_DOT1BR_PORT_TYPE_NONE,
+    SAI_DOT1BR_PORT_TYPE_UPSTREAM,
+    SAI_DOT1BR_PORT_TYPE_CASCADE,
+    SAI_DOT1BR_PORT_TYPE_ACCESS,
+} sai_dot1br_port_type_t;
+
+/**
+ * @brief SAI attributes for SAI_OBJECT_TYPE_DOT1BR_PORT */
+typedef enum _sai_dot1br_port_attr_t
+{
```

Comment:
Dell - Internal Use - Confidential
Hi Atit Jain,

I missed it. In the initial versions of the proposal, these dot1br port attributes were part of the port data attributes. One of the review comments is to create a separate dot1br port object containing only the dot1br attributes. While addressing this comment, I missed adding the physical port to which the dot1br port is mapped to.

I will update the proposal. Following will be the change (highlighted in blue) in the file inc/saidot1brport.h


typedef enum _sai_dot1br_port_attr_t

{

    /** READ-WRITE */



   /** The Port to which the 802.1BR Port is mapped to [sai_object_id_t]  (MANDATORY_ON_CREATE|CREATE_ONLY).

     * Applicable only to Physical ports. */

    SAI_DOT1BR_PORT_ATTR_PORT,



    /** 802.1BR Port Type [sai_dot1br_port_type_t]

     * (MANDATORY_ON_CREATE|CREATE_ONLY).

     * Applicable only to Physical ports. */

    SAI_DOT1BR_PORT_ATTR_TYPE,



    /** 802.1BR Port default ECID [sai_uint32_t] (MANDATORY_ON_CREATE|CREATE_AND_SET).

     * This attribute is valid only if the attribute SAI_DOT1BR_PORT_ATTR_TYPE

     * is set to SAI_DOT1BR_PORT_TYPE_ACCESS.

     * ECID to be added on receiving dot1br untagged frames.

     * Applicable only to Physical ports. */

    SAI_DOT1BR_PORT_ATTR_ECID,



    /** 802.1BR Port default PCP [sai_uint8_t] (CREATE_AND_SET) (default to 0).

     * This attribute is valid only if the attribute SAI_DOT1BR_PORT_ATTR_TYPE

     * is set to SAI_DOT1BR_PORT_TYPE_ACCESS.

     * PCP to be added on receiving dot1br untagged frames.

     * Applicable only to Physical ports. */

    SAI_DOT1BR_PORT_ATTR_PCP,



    /** 802.1BR Port default DEI [sai_uint8_t] (CREATE_AND_SET) (default to 0).

     * This attribute is valid only if the attribute SAI_DOT1BR_PORT_ATTR_TYPE

     * is set to SAI_DOT1BR_PORT_TYPE_ACCESS.

     * DEI to be added on receiving dot1br untagged frames.

     * Applicable only to Physical ports. */

    SAI_DOT1BR_PORT_ATTR_DEI,



    /* -- */



    /* Custom range base value */

    SAI_DOT1BR_PORT_ATTR_CUSTOM_RANGE_BASE  = 0x10000000



} sai_dot1br_port_attr_t;




From: Atit Jain [mailto:notifications@github.com]
Sent: Sunday, June 19, 2016 10:59 PM
To: opencomputeproject/SAI <SAI@noreply.github.com>
Cc: Sivasankar, Ravikumar <Ravikumar_Sivasankar@Dell.com>; Author <author@noreply.github.com>
Subject: Re: [opencomputeproject/SAI] IEEE 802.1BR SAI proposal. (#73)


In inc/saidot1brport.h<https://github.com/opencomputeproject/SAI/pull/73#discussion_r67620543>:

> +

> +/**

> + * @brief Attribute data for SAI_DOT1BR_PORT_ATTR_TYPE

> + */

> +typedef enum _sai_dot1br_port_type_t

> +{

> +    SAI_DOT1BR_PORT_TYPE_NONE,

> +    SAI_DOT1BR_PORT_TYPE_UPSTREAM,

> +    SAI_DOT1BR_PORT_TYPE_CASCADE,

> +    SAI_DOT1BR_PORT_TYPE_ACCESS,

> +} sai_dot1br_port_type_t;

> +

> +/**

> + * @brief SAI attributes for SAI_OBJECT_TYPE_DOT1BR_PORT */

> +typedef enum _sai_dot1br_port_attr_t

> +{

How do we map the dot1br_port to actual/physical port.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/opencomputeproject/SAI/pull/73/files/0195d9e2b31e90a439492b9de2588ceeca8d8687#r67620543>, or mute the thread<https://github.com/notifications/unsubscribe/AOoHu6FjV8O4l1hwTaVyoZH4LZqIAVtqks5qNXxXgaJpZM4HpSGE>.